From c31edeef307fed8ac7c966919d43d892e73424eb Mon Sep 17 00:00:00 2001
From: Tony Liu <Bo.Liu@windriver.com>
Date: Tue, 18 May 2010 12:55:44 +0800
Subject: [PATCH 3/7] lsi_app3k: SMP u-boot reserved region

u-boot needs to reserve a memory region in order to send bootup entry address
to secondary cores by boot core in linux kernel. Add new u-boot tag members to
record the physical start addres and size of the reversed memory region.

Signed-off-by: Tony Liu <Bo.Liu@windriver.com>
---
 arch/arm/include/asm/setup.h |    4 ++++
 arch/arm/mach-app/ubootenv.c |    6 ++++++
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/arch/arm/include/asm/setup.h b/arch/arm/include/asm/setup.h
index 1bf34f2..c81dca2 100644
--- a/arch/arm/include/asm/setup.h
+++ b/arch/arm/include/asm/setup.h
@@ -154,6 +154,10 @@ struct tag_app {
 	__u32 ubootenv0_size;
 	__u32 ubootenv1_size;
 	__u32 pci_sm;
+#ifdef CONFIG_SMP
+	__u32 rev_mem_start;
+	__u32 rev_mem_size;
+#endif
 };
 #endif /*CONFIG_ARCH_APP3K*/
 
diff --git a/arch/arm/mach-app/ubootenv.c b/arch/arm/mach-app/ubootenv.c
index 4c290e4..92ceec6 100644
--- a/arch/arm/mach-app/ubootenv.c
+++ b/arch/arm/mach-app/ubootenv.c
@@ -44,6 +44,8 @@ extern unsigned long host_speed;
 unsigned long uboot_size;
 unsigned long ubootenv0_size;
 unsigned long ubootenv1_size;
+unsigned long secondary_boot_pool;
+unsigned long secondary_boot_pool_len;
 
 static struct environment *environment;
 
@@ -123,6 +125,10 @@ static int __init parse_tag_app(const struct tag *tag)
 	uboot_size = tag->u.app.uboot_size;
 	ubootenv0_size = tag->u.app.ubootenv0_size;
 	ubootenv1_size = tag->u.app.ubootenv1_size;
+#ifdef CONFIG_SMP
+	secondary_boot_pool = tag->u.app.rev_mem_start;
+	secondary_boot_pool_len = tag->u.app.rev_mem_size;
+#endif
 
 	return 0;
 }
-- 
1.6.5.2

